docs/src/userguide/language_basics.rst

```diff
@@ -45,61 +48,102 @@ C variable and type definitions
+.. tabs::
 
-See also :ref:`struct-union-enum-styles`
+    .. group-tab:: Pure Python
 
-.. note::
+        .. literalinclude:: ../../examples/userguide/language_basics/struct_union_enum.py
 
-    Structs can be declared as ``cdef packed struct``, which has
-    the same effect as the C directive ``#pragma pack(1)``.
+        .. NOTE:: Currently, Pure Python mode does not support enums.
 
-Declaring an enum as ``cpdef`` will create a :pep:`435`-style Python wrapper::
+    .. group-tab:: Cython
 
-    cpdef enum CheeseState:
-        hard = 1
-        soft = 2
-        runny = 3
+        .. literalinclude:: ../../examples/userguide/language_basics/struct_union_enum.pyx
 
+        See also :ref:`struct-union-enum-styles`
 
+        .. note::
 
-There is currently no special syntax for defining a constant, but you can use
-an anonymous :keyword:`enum` declaration for this purpose, for example,::
+            Structs can be declared as ``cdef packed struct``, which has
+            the same effect as the C directive ``#pragma pack(1)``.
 
-    cdef enum:
-        tons_of_spam = 3
+        Declaring an enum as ``cpdef`` will create a :pep:`435`-style Python wrapper::
 
-.. note::
-    the words ``struct``, ``union`` and ``enum`` are used only when
-    defining a type, not when referring to it. For example, to declare a variable
-    pointing to a ``Grail`` you would write::
+            cpdef enum CheeseState:
+                hard = 1
+                soft = 2
+                runny = 3
+
+        There is currently no special syntax for defining a constant, but you can use
+        an anonymous :keyword:`enum` declaration for this purpose, for example,::
+
+            cdef enum:
+                tons_of_spam = 3
+
+        .. note::
+            the words ``struct``, ``union`` and ``enum`` are used only when
+            defining a type, not when referring to it. For example, to declare a variable
+            pointing to a ``Grail`` you would write::
 
-        cdef Grail *gp
+                cdef Grail *gp
 
-    and not::
+            and not::
 
-        cdef struct Grail *gp # WRONG
+                cdef struct Grail *gp # WRONG
 
-    There is also a ``ctypedef`` statement for giving names to types, e.g.::
+            There is also a ``ctypedef`` statement for giving names to types, e.g.::
 
-        ctypedef unsigned long ULong
+                ctypedef unsigned long ULong
 
-        ctypedef int* IntPtr
+                ctypedef int* IntPtr
 
 
 It is also possible to declare functions with :keyword:`cdef`, making them c functions.
 
-::
+.. tabs::
 
-    cdef int eggs(unsigned long l, float f):
-        ...
+    .. group-tab:: Pure Python
+
+        .. code-block:: python
+
+            @cython.cfunc
+            def eggs(l: cython.ulong, f: cython.float) -> cython.int:
+                ...
+
+    .. group-tab:: Cython
+
+        .. code-block:: cython
+
+            cdef int eggs(unsigned long l, float f):
+                ...
```

Comment:
The versions will not align here, it is easier to achieve by moving the code out to the file and adding a space to the line before the `cdef` line.